### PR TITLE
Feature/sort by event start date

### PR DESCRIPTION
**Pull request by frank3manuel**
FEATURE:
 - Sort topics of event categories by event start date. Event categories have custom field `sort_topics_by_event_start_date` set to true.
 - Disable re-sorting of categories with custom field `disable_topic_resorting`
 
 Tests to follow.

### CODE REVIEW

File: assets/javascripts/discourse/adapters/connectors/category-custom-settings/show-event-category-sorting-settings.hbs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{{!-- {{this.category.custom_fields.sort_topics_by_event_start_date}} --}}
```
Comment (ZogStriP) [inline assets/javascripts/discourse/adapters/connectors/category-custom-settings/show-event-category-sorting-settings.hbs:1]: Is this comment a test/debug leftover?

Comment (frank3manuel) [reply assets/javascripts/discourse/adapters/connectors/category-custom-settings/show-event-category-sorting-settings.hbs:1]: Ah yes, I'll remove it.

File: test/javascripts/acceptance/sort-event-topics-test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { acceptance, exists } from "discourse/tests/helpers/qunit-helpers";
+import { test } from "qunit";
+import { visit } from "@ember/test-helpers";
+import { cloneJSON } from "discourse-common/lib/object";
+import CategoryFixtures from "discourse/tests/fixtures/category-fixtures";
+
+acceptance("Calendar - Disable sorting headers", function (needs) {
+  needs.user();
+  needs.pretender((server, helper) => {
+    const cardResponse = cloneJSON(CategoryFixtures["/c/1/show.json"]);
```
Comment (jbrw) [inline test/javascripts/acceptance/sort-event-topics-test.js:10]: The _smallest_ of suggestions: `cardResponse` probably should be `categoryResponse` or similar.

Comment (frank3manuel) [reply test/javascripts/acceptance/sort-event-topics-test.js:10]: You're right. Thanks.

**Review by jbrw — APPROVED** [review]

File: assets/javascripts/initializers/disable-sort.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import discourseComputed from "discourse-common/utils/decorators";
+import { withPluginApi } from "discourse/lib/plugin-api";
+
+export default {
+  name: "disable-sort",
+  initialize() {
+    withPluginApi("0.8", (api) => {
+      api.modifyClass("component:topic-list", {
```
Comment (CvX) [inline assets/javascripts/initializers/disable-sort.js:8]: @frank3manuel This `api.modifyClass()` is missing `pluginId` which results in warnings

```js
api.modifyClass("component:topic-list", {
  pluginId: "discourse-calendar",

  …
});
```

Comment (frank3manuel) [reply assets/javascripts/initializers/disable-sort.js:8]: Will fix this, thanks.

Comment (CvX) [reply assets/javascripts/initializers/disable-sort.js:8]: Ha, too late! 😛 https://github.com/discourse/discourse-calendar/pull/325

File: assets/javascripts/initializers/disable-sort.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        @discourseComputed("category")
+        sortable(category) {
+          let disableSort = true;
```
Comment (CvX) [inline assets/javascripts/initializers/disable-sort.js:11]: @frank3manuel a bug spotted by @keegangeorge – sorting topics doesn't work in e.g. /latest

I believe that was supposed to be `false` by default?

Comment (frank3manuel) [reply assets/javascripts/initializers/disable-sort.js:11]: If this hasn't been fixed yet, I'll fix it now.

Comment (frank3manuel) [reply assets/javascripts/initializers/disable-sort.js:11]: PR for the fix is up. #334

**Pull request merged**